src/components/ContactsList/index.js

```diff
@@ -0,0 +1,144 @@
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Benjamin", familyName: "André" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Bertrand", familyName: "Alevin" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "André", familyName: "Manoukian" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Erika", familyName: "Shell" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Brendan", familyName: "Abolivier" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "COZY", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "EDF", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "CAF", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Eric", familyName: "D" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  }
+];
+
+export const contactSort = (contact, comparedContact) => {
+  const firstNameA =
+    contact.name && contact.name.givenName
+      ? contact.name.givenName.toUpperCase()
+      : "";
+  const firstNameB = comparedContact.name
+    ? comparedContact.name.givenName.toUpperCase()
+    : "";
+
+  if (firstNameB === "") return -1;
+  if (firstNameA === "") return 1;
+  if (firstNameA < firstNameB) return -1;
+
+  if (firstNameA > firstNameB) return 1;
+
+  const lastNameA =
+    contact.name && contact.name.familyName
+      ? contact.name.familyName.toUpperCase()
+      : "";
+  const lastNameB =
+    comparedContact.name && comparedContact.name.familyName
+      ? comparedContact.name.familyName.toUpperCase()
+      : "";
+  if (lastNameA < lastNameB) return -1;
+  if (lastNameA > lastNameB) return 1;
+  return 0;
+};
+
+export const withContacts = BaseComponent =>
+  class WithContacts extends Component {
+    state = {
+      contacts: sampleContacts
```

Comment:
Do you want this to be part of the final PR? If I understood correctly, we wanted to have a loading state instead of placeholder contacts.

Reply:
If I remove this `sampleContacts`, everybody have to add data to get a screen.
When cozy-client will be added, this HOC will disappears, and I hope developers will also remove the sampleContacts array.

Do you want I remove the array, or are we good?